add quartz dependency

IvanLinnik opened the pull request:
add scheduler folder with quartz.properties, scheduler and job classes

it's just for testing for now. tell me if i'm on the wrong way.

IvanLinnik committed:
add quartz dependency
add scheduler folder with quartz.properties, scheduler and job classes

it's just for testing for now. tell me if i'm on the wrong way.

mmarusich commented on build.gradle:
```diff
@@ -25,6 +25,9 @@ dependencies {
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
+
+    // https://mvnrepository.com/artifact/org.quartz-scheduler/quartz
+    implementation group: 'org.quartz-scheduler', name: 'quartz', version: '2.3.2'
```
use starter from spring boot

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/scheduler/CheckUpdateScheduler.java:
```diff
@@ -0,0 +1,27 @@
+    public static void main(String[] args) {
+        try {
+            Scheduler scheduler = StdSchedulerFactory.getDefaultScheduler();
+            scheduler.start();
```
if you will use starter, spring boot auto configuration will do this job

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/scheduler/quartz.properties:
```diff
@@ -0,0 +1,3 @@
+org.quartz.scheduler.instanceName = StreamingServiceScheduler
```
don't create separate properties file, place this settings into regular application.properties file

IvanLinnik committed:
change dependency on spring-boo-starter-quartz
move quartz.properties into application.properties

IvanLinnik committed:
Merge branch 'master' into add_quartz_scheduler

IvanLinnik committed:
add SpringQuartzConfig.java
remove version in build.gradle
rewrite CheckUpdateJob

LinnikIvan committed:
Merge branch 'master' of https://github.com/IvanLinnik/InternshipStreamingServiceParserXML into add_quartz_scheduler

 Conflicts:
	src/main/resources/application.properties

IvanLinnik committed:
pull changes from master
add SpringQuartzConfig.java
remove version in build.gradle
rewrite CheckUpdateJob

LinnikIvan committed:
Merge branch 'add_quartz_scheduler' of https://github.com/IvanLinnik/InternshipStreamingServiceParserXML into add_quartz_scheduler

 Conflicts:
	src/main/resources/application.properties

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/scheduler/SpringQuartzConfig.java:
```diff
@@ -0,0 +1,32 @@
+    public JobDetail jobDetail() {
+        return JobBuilder.newJob().ofType(CheckUpdateJob.class)
+                         .storeDurably()
+                         .withIdentity("Updater")
```
use job key for identity,

example https://github.com/kenshoo/fb-product-fetcher/blob/master/application/src/main/java/io/skai/fbproductfetcher/quartz/config/QuartzJobConfig.java#L19

IvanLinnik replied:
done

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/scheduler/SpringQuartzConfig.java:
```diff
@@ -0,0 +1,32 @@
+    @Bean
+    public Trigger trigger(JobDetail job) {
+        return TriggerBuilder.newTrigger().forJob(job)
+                             .withIdentity("Time_Trigger")
```
same

IvanLinnik replied:
done

IvanLinnik committed:
add JobKey and TriggerKey
